pkg/build/builder/docker.go

```diff
@@ -145,13 +144,7 @@ func (d *DockerBuilder) checkSourceURI() error {
 			host += ":443"
 		}
 	}
-	dialer := net.Dialer{Timeout: d.urlTimeout}
-	conn, err := dialer.Dial("tcp", host)
-	if err != nil {
-		return err
-	}
-	return conn.Close()
-
```

Comment:
method comment should change
port defaulting can go away (since the result isn't used any more)
probably shouldn't prepend `http://` or `https://` any more (that could turn valid clone specs like `ssh://foo...` invalid, e.g. `https://ssh://...`)